packages/auth/amplify_auth_cognito_dart/lib/src/sdk/sdk_bridge.dart

```diff
@@ -795,77 +797,71 @@ extension MfaSettings on CognitoIdentityProviderClient {
     required String accessToken,
     MfaPreference? sms,
     MfaPreference? totp,
+    MfaPreference? email,
   }) async {
```

Comment:
nit: This function is getting hard to read. Breaking the logic down into helper functions would improve readability.
_calculatePreferredMethod()
_generateUserMfaPreferenceRequest()

Reply:
Please refactor this, it is really hard to read and understand the logic.